### DISCUSSION TITLE
How to reset Thumbnails

### DISCUSSION THREAD

**Discussion by findnix** [rern/rAudio#647] [Fixed Issues]
Hallo Friends of rAudio!

Since a while a lot of thumbnails where gone and I can see only the small but correct icon in the directory view.
One step deeper, in the "Album" view, there is only the default "rAudio" picture, where i already could see the Album picture.
Same for the player: vu-meter or default picture.

A simple "Update thumbnails" in the entry point of the music directory tree won't help.   
Is there a way to reset and rebuild the "Cover DB"?

Many thanks in advance
Ronald

**Comment by klassikmann** [comment]
@findnix 
> Is there a way to reset and rebuild the "Cover DB"?

Yes.
There a three kinds of coverarts:
1. The "real" big coverart, say "folder.jpg", supplied by the user and presented in the playback view
2. The medium size coverart, named "coverart.jpg" generated by rAudio for use in the album view
3. The tiny coverart, named "thumb.jpg" generated by rAudio for use in the playlist view

To rebuild the "Cover DB" remove all "coverart.jpg" and "thumb.jpg" and then perform an "Update thumbnails".

**Reply by findnix** [reply]
Thank You @klassikmann !

I did a `find . -name .. -exec rm ..`  for coverart.jpg and thumb.jpg and an "Update Thumbnails".
Saw a lot of "creating thumbnail" messages.
Now the directory icons are OK, but still no individual cover in album and player view. 

Also cleared browser cache, restated chrome, tested with ms-edge too and rebooted the RPi. 
Checked new generated jpgs: OK in different sizes and readable.

Do You have any other idea to check?

LG.R

**Reply by klassikmann** [reply]
@findnix 
> Do You have any other idea to check?

Hmm.
Do I understand correctly? The thumb(s).jpg and coverart(s).jpg will be shown but not the folder(s).jpg? If yes, perhaps I have an idea:
FWIW: Today I made the experience that the folder(s).jpg in the playback window were missing. I had a look at the "Playback Display" settings: all OK, only "WebRadio time" and "Coverart" (but slightly dimmed) are set. I started to play around at the settings, checking/unchecking "Top-Bottom bars", "Time", ... and, all of  sudden, the full folder image was available again. I didn't invesigate any further. 
A bug or not a bug, that is the question.

**Comment by findnix** [comment]
compared a lot an found something funny:
i've two disks for my music library: an internal m.2 ssd with ext4 and an external usb disk with ntfs, mounted with the fuse driver:
```
/dev/sda1 on /mnt/MPD/USB/1.42.6-25426 type ext4 (rw,nosuid,nodev,noexec,noatime)
/dev/sdb1 on /mnt/MPD/USB/rb-1TB type fuseblk (rw,nosuid,nodev,noexec,noatime,user_id=0,group_id=0,default_permissions,allow_other,blksize=4096)
```
checked permissions and owner: ident
```
rAudio-VIE:/mnt/MPD/USB  #  ls -l
total 68
drwxrwxrwx 5 root root 4096 Jan 24  2021 1.42.6-25426
drwxrwxrwx 1 root root 4096 Jan 30  2021 rb-1TB
```
 
and now the strange thing:
all cover arts in NTFS are displayed correctly,
but not in EXT4.
even did a fsck.ext4

```
rAudio-VIE:/mnt/MPD/USB/1.42.6-25426/music/0000 Africa [mp3+FLAC]/Ali Farka Toure with Ry Cooder [mp3+FLAC]/Talking Timbuktu [1994] [FLAC-44.1-16]  #  ls -l
total 329836
-rwxrwxrwx 1 root root 30334751 Aug 12  2015 '01- Bonde.flac'
-rwxrwxrwx 1 root root 36210177 Aug 12  2015 '02- Soukora.flac'
-rwxrwxrwx 1 root root 37036811 Aug 12  2015 '03- Gomni.flac'
-rwxrwxrwx 1 root root 16361714 Aug 12  2015 '04- Sega.flac'
-rwxrwxrwx 1 root root 50427523 Aug 12  2015 '05- Amandrai.flac'
-rwxrwxrwx 1 root root 40791665 Aug 12  2015 '06- Lasidan.flac'
-rwxrwxrwx 1 root root 30927454 Aug 12  2015 '07- Keito.flac'
-rwxrwxrwx 1 root root 12905021 Aug 12  2015 '08- Banga.flac'
-rwxrwxrwx 1 root root 42060708 Aug 12  2015 '09- Ai Du.flac'
-rwxrwxrwx 1 root root 37860309 Aug 12  2015 '10- Diaraby.flac'
drwxrwxrwx 2 root root     4096 Mar 28  2016  Artwork
-rwxrwxrwx 1 root root  1377299 Aug 12  2015 'a Front.jpg'
-rwxrwxrwx 1 root root  1377299 Nov 22 18:24  cover.jpg
-rw-r--r-- 1 root root    43237 Nov 22 22:09  coverart.jpg
-rw-r--r-- 1 root root     7979 Nov 22 22:09  thumb.jpg
rAudio-VIE:/mnt/MPD/USB/1.42.6-25426/music/0000 Africa [mp3+FLAC]/Ali Farka Toure with Ry Cooder [mp3+FLAC]/Talking Timbuktu [1994] [FLAC-44.1-16]  #
```

![image](https://user-images.githubusercontent.com/77154016/142938265-a7aa6d52-e8e3-4fa4-898f-da48a0be21c0.png)
![image](https://user-images.githubusercontent.com/77154016/142938333-ece517fb-0271-4e49-9553-d6a263c1f72f.png)
![image](https://user-images.githubusercontent.com/77154016/142938405-f4a9b25e-9fde-47e2-95ec-f1b2013a7482.png)
![image](https://user-images.githubusercontent.com/77154016/142938473-dea7b72d-51a3-436d-8db4-37ac876af7ad.png)

**Reply by klassikmann** [reply]
I had a problem in the 20211121 release with images of bookmarks. They were not displayed for albums on a NAS (resolved in 20211122). Looks similar.
I think, it's a task for @rern now.

**Reply by findnix** [reply]
Thank you @klassikmann, I appreciate your support.
Since the sound quality is untouched, I will exercise patience :-)
@rern Feel free to request logfiles. It is my honor to assist in the troubleshooting process.

**Reply by rern** [reply]
Try:
```sh
sed -i 's|coverfile/\.\*|coverfile:0:-4|; s|coverfile/\*\.|coverfile: -3|' /srv/http/bash/status-coverart.sh
```

**Reply by findnix** [reply]
Thank You @rern!
Done!
```
 diff /srv/http/bash/status-coverart.sh /srv/http/bash/status-coverart.sh.ori
21c21
<       [[ -e $dirshm/online/$covername.$ext ]] && echo ${coverfile:0:-4}.$date.$ext && exit
---
>       [[ -e $dirshm/online/$covername.$ext ]] && echo ${coverfile/.*}.$date.$ext && exit
32c32
<       echo ${coverfile:0:-4}.$date.${coverfile: -3} | tee $dirshm/local/$covername
---
>       echo ${coverfile/.*}.$date.${coverfile/*.} | tee $dirshm/local/$covername

```
No immediate visibel changes.
Removed coverart.jpg and thumb.jpg and rescanned. No change in behavior.
What owner and access pattern do you expect for the "music tree? 
It looks like this:
```
rAudio-VIE:/mnt/MPD/USB/1.42.6-25426/music/0000 Africa [mp3+FLAC]/Ali Farka Toure with Ry Cooder [mp3+FLAC]  #  ls -lR
.:
total 8
drwxrwxrwx 3 mpd audio 4096 Nov 23 12:11 'Talking Timbuktu [1994] [FLAC-44.1-16]'
drwxrwxrwx 3 mpd audio 4096 Nov 22 22:31 'Talking Timbuktu [1994] [mp3@320]'

'./Talking Timbuktu [1994] [FLAC-44.1-16]':
total 329836
-rwxrwxrwx 1 mpd  audio 30334751 Aug 12  2015 '01- Bonde.flac'
-rwxrwxrwx 1 mpd  audio 36210177 Aug 12  2015 '02- Soukora.flac'
-rwxrwxrwx 1 mpd  audio 37036811 Aug 12  2015 '03- Gomni.flac'
-rwxrwxrwx 1 mpd  audio 16361714 Aug 12  2015 '04- Sega.flac'
-rwxrwxrwx 1 mpd  audio 50427523 Aug 12  2015 '05- Amandrai.flac'
-rwxrwxrwx 1 mpd  audio 40791665 Aug 12  2015 '06- Lasidan.flac'
-rwxrwxrwx 1 mpd  audio 30927454 Aug 12  2015 '07- Keito.flac'
-rwxrwxrwx 1 mpd  audio 12905021 Aug 12  2015 '08- Banga.flac'
-rwxrwxrwx 1 mpd  audio 42060708 Aug 12  2015 '09- Ai Du.flac'
-rwxrwxrwx 1 mpd  audio 37860309 Aug 12  2015 '10- Diaraby.flac'
drwxrwxrwx 2 mpd  audio     4096 Mar 28  2016  Artwork
-rwxrwxrwx 1 mpd  audio  1377299 Aug 12  2015 'a Front.jpg'
-rwxrwxrwx 1 mpd  audio  1377299 Nov 22 18:24  cover.jpg
-rw-r--r-- 1 root root     43237 Nov 23 12:11  coverart.jpg
-rw-r--r-- 1 root root      7979 Nov 23 12:11  thumb.jpg
[...]

```

**Reply by rern** [reply]
Next: `rm /srv/http/data/shm/local/*`

**Reply by findnix** [reply]
Thank You!

There was a lot of garbage in this directory.
Worked immediately. There was no additional "Update thumbnails" run necessary.

Thank You so much again.

BTW: 
I can't find a donation button. 
Pls give me the change to support you and your team.

**Reply by ghost** [reply]
Hi,

I‘m running rAudio-1 version 202204151 on a RasPi3B+ and there is one 'cover.jpg' (600x600) per album folder so playback view works. But there is no coverart.jpg and no thumbnail.jpg. 

I‘ve also tested soundirok (command „ ln -s /mnt/MPD /srv/http/music“). Within the program it works but I prefer the web gui. Renaming to 'folder.jpg', clearing cache, rebuilding database/update thumbnail doesn't help. 

A kind of 'workaround': renaming all 'cover.jpg' to 'coverart.jpg' works in album view and play mode but still no thumbnails.

/srv/http/data/shm/local/* is empty.

Anything else?

Thanks and best regards,
Oliver

==============

[edit]: I got it! I‘ve never thought about my NAS audio user permissions because it wasn‘t necessary…missing write option. 😉

Renaming all cover.jpg to folder.jpg, complete db rescan, update thumbnail…and now I‘m very happy with it! 👍🏻
